File: trasposta.py
#! Calculate transposed matrix of a generic 2D matrix.

def transpose_function (matrix):
    row = len (matrix)                                  # Number of rows
    col = len (matrix[0])                               # Number of columns
    
    # Create the space for the transposed matrix (full of zeros)
    transposed = []
    for i in range (col):
        transposed_row = [0] * col
        transposed.append (transposed_row)
    
    # Insert the values in the transposed matrix
    for i in range (row):
        for j in range (col):
            transposed[j][i] = matrix[i][j]             # transposed has rows and columns inverted
    
    return transposed



def main ():

    matrix = [
        [1, 2, 3],
        [4, 5, 6],
        [7, 8, 9]
    ]

    transposed_matrix = transpose_function (matrix)

    # Print transposed matrix
    print ("Transposed matrix:\n")
    for row in transposed_matrix:
        print (row)
    
    return 0

if __name__ == "__main__":
    main()

File: trasposta.c
//! Calculate transposed matrix of a generic 2D matrix.

#include <stdio.h>

void transpose_function(int matrix[][3], int row, int col, int transposed[][3]) {
    // Create the space for the transposed matrix (full of zeros)
    for (int i = 0; i < col; i++) {
        for (int j = 0; j < col; j++) {
            transposed[i][j] = 0;
        }
    }

    // Insert the values in the transposed matrix
    for (int i = 0; i < row; i++) {
        for (int j = 0; j < col; j++) {
            transposed[j][i] = matrix[i][j]; // transposed has rows and columns inverted
        }
    }
}

int main() {
    int matrix[3][3] = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };
    int transposed_matrix[3][3];

    int row = 3; // Number of rows
    int col = 3; // Number of columns

    // Calculate transposed matrix
    transpose_function(matrix, row, col, transposed_matrix);

    // Print transposed matrix
    printf("Transposed matrix:\n");
    for (int i = 0; i < col; i++) {
        for (int j = 0; j < row; j++) {
            printf("%d ", transposed_matrix[i][j]);
        }
        printf("\n");
    }

    return 0;
}
